lib/ecto/multi.ex

```diff
@@ -242,23 +242,27 @@ defmodule Ecto.Multi do
     delete(multi, name, Changeset.change(struct), opts)
   end
 
-  defp add_changeset(multi, action, name, changeset, opts) when is_list(opts) do
+  defp add_changeset(multi, action, name, %Changeset{} = changeset, opts) when is_list(opts) do
```

Comment:
actually, I'll revert this change so that `put_action` can respond with the right error when this somehow gets invalid argument.